src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -972,6 +976,40 @@
               :subquery? false
               :source-table "producers"}))
 
+(def packages-query
+  "Basic packages query"
+  (map->Query {:projections {"certname" {:type :string
+                                         :queryable? true
+                                         :field :certnames.certname}
+                             "package_name" {:type :string
+                                             :queryable? true
+                                             :field :pi.name}
+                             "version" {:type :string
+                                        :queryable? true
+                                        :field :pi.version}
+                             "provider" {:type :string
+                                         :queryable? true
+                                         :field :pi.provider}}
+
+               :selection {:from [[:package_inventory :pi]]
+                           :left-join [:certnames
+                                       [:= :pi.certname_id :certnames.id]]}
```

Comment:
we talked about this and came to the conclusion that, since (a) this can only really go wrong if you start deleting stuff yourself and (b) we have pretty good fk constraints, we're probably ok as-is.